lib/addoninfo.cpp

```diff
@@ -52,47 +52,70 @@ static std::string parseAddonInfo(AddonInfo& addoninfo, const picojson::value &j
         return "Loading " + fileName + " failed. " + json_error;
     }
     if (!json.is<picojson::object>())
-        return "Loading " + fileName + " failed. Bad json.";
-    picojson::object obj = json.get<picojson::object>();
-    if (obj.count("args")) {
-        if (!obj["args"].is<picojson::array>())
-            return "Loading " + fileName + " failed. args must be array.";
-        for (const picojson::value &v : obj["args"].get<picojson::array>())
-            addoninfo.args += " " + v.get<std::string>();
-    }
+        return "Loading " + fileName + " failed. JSON is not an object.";
 
-    if (obj.count("ctu")) {
-        // ctu is specified in the config file
-        if (!obj["ctu"].is<bool>())
-            return "Loading " + fileName + " failed. ctu must be boolean.";
-        addoninfo.ctu = obj["ctu"].get<bool>();
-    } else {
-        addoninfo.ctu = false;
+    // TODO: remove/complete default value handling for missing fields
+    const picojson::object& obj = json.get<picojson::object>();
+    {
+        const auto it = obj.find("args");
+        if (it != obj.cend()) {
+            const auto& val = it->second;
+            if (!val.is<picojson::array>())
+                return "Loading " + fileName + " failed. 'args' must be an array.";
+            for (const picojson::value &v : val.get<picojson::array>())
+                addoninfo.args += " " + v.get<std::string>();
```

Comment:
old problem.. but we should check that the value is a string here.

Reply:
Done.

Reply:
Isn't it time to implement JSON schema validation or similar? I didn't do any research to speak of but there must be some JSON schema format that can be used from C++ and Python. The schemas could serve as documentation and the code can safely assume that all JSON types are correct.

Reply:
I also never used JSON schemas. Currently still looking into making everything const (and possibly even improving the performance of addon invocation a bit along the way).

I was thinking about having some serialization code but that obviously requires reflection.

We do use schemas for our XMLs so that definitely makes sense.

Reply:
Let's discuss! https://github.com/danmar/cppcheck/pull/5852